### ISSUE TITLE
Error on mint in production

### ISSUE THREAD

**Issue by arkitoure** [wearekite/candy-machine-v3-ui#7]
Hello @wearekite! First off thx for your efforts on this, best implementation atm.

Im having an issue when deploying in Vercel. Mint is configured correctly but when going to mint an NFT transaction always fails to simulate and mint. Getting a strange error in the console that perhaps you might have seen before?

> MetaplexError: CandyGuardProgram > Unknown Program Error
> >> Source: Program > CandyGuardProgram [Guard1JwRhJkVH6XZhzoYxeBVQe872VH6QggF4BWmS9g]
> >> Problem: The program [CandyGuardProgram] at address [Guard1JwRhJkVH6XZhzoYxeBVQe872VH6QggF4BWmS9g] raised an error that is not recognized by the programs registered by the SDK.
> >> Solution: Unfortunately, you will need to check the unparsed error below to investigate what went wrong. To get more helpful error messages, ensure the program that failed is registered by the SDK and provides an "errorResolver" method.
> 
> Caused By: Error: failed to send transaction: Transaction simulation failed: Error processing Instruction 4: custom program error: 0x17a3

It seems like the candy machine ID does not read and returns this error. Since candy guard is engaged, the authority of the candy machine changes to the guard. Could this be the cause of this error? Is there an extra step required to authorize minting on the front end through the candy guard authority?

**Comment by arkitoure** [comment]
Hey thx @wearekite. Message on the way.

**Comment by mariospeterman** [comment]
Hi there,

First of all, I want to express my appreciation for the incredible work and effort you've put into this project for the community. I am, however, experiencing the same issue as discussed earlier in this thread. It appears that using Sugar CLI > 2.0.0 creates a Candy Machine v3 with a MintV1 account, which is causing the error and suggesting upgrading to a MintV2 account. I would like to know how you set up your Candy Machine v3.

@arkitoure @wearekite, could you please share which Sugar CLI version you used or if you set up the Candy Machine v3 using a different library?

Thank you in advance for your help!

**Comment by arkitoure** [comment]
Hello @mariospeterman ... we are currently working on this issue with several tests to see if we can repeat it. On my side, we are also using SugarCLI/Rust but will run on with the JS Umi library to see the result. Will be posting the outcomes here today.

Using the most recent SugarCLI crate:
https://crates.io/crates/sugar-cli

**Comment by mariospeterman** [comment]
Hi @arkitoure I have tried different versions of Sugar CLI, but unfortunately, none of them have worked for me. I am looking forward to the results of your tests and hoping that the Umi library might provide a solution to this issue.

**Comment by arkitoure** [comment]
Hi @mariospeterman Just to add, I might just skip ahead to the Umi library now for a test since your explanation is detailed enough for me to believe this may be a bug with the SugarCLI. I was about to do another SugarCLI test on the devnet but if you are experiencing the same issue we already have a repeated similar result.

Can you do one thing though? Run "sugar show" cmd on the collection you are having an issue with and post the output of the "account version"?

**Comment by wearekite** [comment]
Hey @mariospeterman & @arkitoure 

Seems that you are using the regular Sugar CLI, not the version built for Candy Machine which is currently still in alpha.
I am using the latest [Sugar CMv3 CLI](https://github.com/metaplex-foundation/sugar/releases/tag/sugar-cmv3-alpha.3), which works. Would explain the issue at hand.

If you're on Windows, you can download the .exe file and reference it in your project like `./sugar.exe -V`
![Screenshot 2023-04-29 142131](https://user-images.githubusercontent.com/100683095/235302300-871c07ae-e1b6-4326-9889-97ca91f19595.png)

Let me know if this has resolved the issues at hand so I can close the issue 👍

**Comment by arkitoure** [comment]
Hi @wearekite ...hmm, this is slightly confusing. Does this mean the most recent update (a month ago) to the SugarCLI is another version to this alpha from Dec.? I have to look deeper into the code base so conflicts are not created on the file dependencies. Things can get a little tricky when running multiple versions of the same thing. Let me run some quick tests, brb.

**Comment by wearekite** [comment]
Hey @arkitoure,

Completely understandable. The current documentation about sugar is pretty two-sided, the one article mentioning SugarV2 also supports CMv3, while the other mentions Sugar is only for CMv2 unless it has the cmv3 prefix then it suppport CMv3.

I am going off my personal knowledge with Sugar and having read their documentation through and through, so I could very well be wrong here. But that Sugar CLI currently only supports a different Mint version than their SugarCMv3 CLI.

This would explain why my CMv3 UI breaks with newer (or/and not cmv3 specific) Sugar versions.

To support my argument: See [article 1](https://docs.metaplex.com/developer-tools/sugar/overview/quick-start), a tutorial for launching a candy machine with the regular Sugar. If you scroll to the bottom of that article you will find a link that leads to their [CMv2 UI](https://docs.metaplex.com/guides/candy-machine-ui). For which I would think that that article is a guide for creating a CMv2 with regular Sugar.

To even better support my argument & where I first learnt Sugar for CMv3 is [article 2](https://docs.metaplex.com/programs/candy-machine/how-to-guides/my-first-candy-machine-part1) where they mention to install the alpha version of sugar; sugar-cmv3
![image](https://user-images.githubusercontent.com/100683095/235303503-ff5935c9-fa50-4a2e-9c8f-7685f94b459d.png)

Test it out, and see if it works. Sugar is still new, so you might find documentation that is outdated so always be sure to check when the articles are updated, you can find when at the bottom of each article. 

Example: This is article (article 2 linked in this comment) was updated just 3 days ago on Apr 26th.
![image](https://user-images.githubusercontent.com/100683095/235303576-47004ff0-2884-48fb-ad69-96609d034ecd.png)

**Comment by arkitoure** [comment]
Ok looked deeper into the code and it looks like the current version is accurate as it was merged into SugarCLI v2beta1. Check the branch merge here: https://github.com/metaplex-foundation/sugar/compare/sugar-cmv3-alpha.3...v2.0.0-beta.1

So the current version (v2.1.1) should be ok to use for CMv3 deployments. Im still going to move on with my tests now anyway just to eliminate any possibilities.

**Comment by wearekite** [comment]
> Ok looked deeper into the code and it looks like the current version is accurate as it was merged into SugarCLI v2beta1. Check the branch merge here: [metaplex-foundation/sugar@sugar-cmv3-alpha.3...v2.0.0-beta.1](https://github.com/metaplex-foundation/sugar/compare/sugar-cmv3-alpha.3...v2.0.0-beta.1)
> 
> So the current version (v2.1.1) should be ok to use for CMv3 deployments. Im still going to move on with my tests now anyway just to eliminate any possibilities.

Quickly checked for myself too. What I found is that only their SugarV2-beta releases are merged with SugarV1.1.2-alpha-cmv3, not their newer versions
Their base (stable version) v2.0.0
![image](https://user-images.githubusercontent.com/100683095/235304058-b83a98d7-7eaa-4a2b-869d-cf58037bfa55.png)

and their v2.0.0-beta.1&2
![image](https://user-images.githubusercontent.com/100683095/235304122-b9ba60fd-7e43-4293-b248-8e9610bb1dcb.png)

So seems to me that using their newest Sugar versions still don't support CMv3 fully, but their v2.0.0-beta versions do.

**Comment by wearekite** [comment]
Commenting to update on a quick test that I have run with SugarV2-beta.2

All went well, mint worked and showed up in wallet on the latest deployment of my UI

Will repeat test with SugarV2.1.1

**Comment by wearekite** [comment]
SugarV2.1.1 test also completed:
![image](https://user-images.githubusercontent.com/100683095/235305540-20eabc3b-6646-4eb4-b9fb-713299c39f74.png)

This one caused the same error stated in the opening message of this issue. Meaning it did **not** work. 

TLDR; Use [v2.0.0-beta.2](https://github.com/metaplex-foundation/sugar/releases/tag/v2.0.0-beta.2) (latest as of posting this message) which supports CMv3 fully and so this UI.

**Comment by arkitoure** [comment]
Nice stuff! Thx for the deep dive on this. Ive been held up temp but I will post my results soon here just to round things off.

**Comment by arkitoure** [comment]
@wearekite thx for pinpointing this issue. The rollback of version has worked. UI now functions as expected.

To those who use the Rust Crate of the SugarCLI, safe rollback was as follows:

> cargo uninstall -p sugar-cli -v
> cargo install --version 2.0.0-beta.2 sugar-cli -v

(remove the optional  -v, verbose flag, if desired)
